src/reconcile/rec_visibility.c

```diff
@@ -222,6 +222,54 @@ __rec_need_save_upd(
         return (!__wt_txn_tw_start_visible_all(session, &upd_select->tw));
 }
 
+/*
+ * __get_valid_upd --
+ *     Loop until a valid update from a different transaction is found in the update list. As a side
+ *     effect method saves the latest update from the same transaction into same_txn_valid_upd.
+ *     Returns a valid update from a different transaction
+ */
+static inline WT_UPDATE *
+__get_valid_upd(WT_UPDATE *upd, WT_UPDATE *tombstone, WT_UPDATE **same_txn_valid_upd)
+{
+    while (upd->next != NULL) {
+        if (upd->next->txnid == WT_TXN_ABORTED)
+            upd = upd->next;
+        else if (upd->next->txnid != WT_TXN_NONE && tombstone->txnid == upd->next->txnid) {
+            upd = upd->next;
+            /* Save the latest update from the same transaction. */
+            if (*same_txn_valid_upd == NULL)
+                *same_txn_valid_upd = upd;
+        } else
+            break;
+    }
+
+    return upd;
+}
+
+/*
+ * __timestamp_sanity_check --
+ *     If we found a tombstone with a time point earlier than the update it applies to, which can
+ *     happen if the application performs operations with timestamps out-of-order, make it invisible
+ *     by making the start time point match the stop time point of the tombstone. We don't guarantee
+ *     that older readers will be able to continue reading content that has been made invisible by
+ *     out-of-order updates. Note that we carefully don't take this path when the stop time point is
+ *     equal to the start time point. While unusual, it is permitted for a single transaction to
+ *     insert and then remove a record. We don't want to generate a warning in that case.
+ */
+static inline void
+__timestamp_sanity_check(WT_SESSION_IMPL *session, WT_TIME_WINDOW *select_tw, char *time_string)
```

Comment:
Instead of `sanity` check how about adding mentioning `out of order` itself in the function name? Like `__timestamp_outoforder_fix`? I am not worried much about it, but `check` may not sound good as it is correcting the values.

Reply:
Renamed to __timestamp_out_of_order_fix. Thanks!